src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java

```diff
@@ -389,14 +393,24 @@ private FedoraResource getBaseVersion(final FedoraResource resource) {
                     final Map<String, List<String>> aclTriples = new HashMap<>();
                     child.getTriples(translator, PROPERTIES)
                         .filter(p -> isAclPredicate.test(model.asStatement(p).getPredicate()))
-                        .forEach(t -> {
-                            aclTriples.putIfAbsent(t.getPredicate().getURI(), new ArrayList<>());
-                            if (t.getObject().isURI()) {
-                                aclTriples.get(t.getPredicate().getURI()).add(
-                                    substringBeforeLast(t.getObject().getURI(), "/" + JCR_CONTENT));
-                            } else if (t.getObject().isLiteral()) {
-                                aclTriples.get(t.getPredicate().getURI()).add(
-                                    t.getObject().getLiteralValue().toString());
+                        .forEach(triple -> {
+                            final String predicate = triple.getPredicate().getURI();
+                            aclTriples.putIfAbsent(predicate, new ArrayList<>());
+                            if (triple.getObject().isURI()) {
+                                final String object = triple.getObject().getURI();
+                                if (predicate.equals(WEBAC_AGENT_VALUE)) {
+                                    if (userBaseUri != null && object.startsWith(userBaseUri)) {
+                                        aclTriples.get(predicate).add(object.substring(userBaseUri.length()));
```

Comment:
As discussed, we should wait on use cases, but in the meantime, had you considered using the `object.toString()` approach for URIs instead of assuming a single, common baseUri?